mmengine/dist/utils.py

```diff
@@ -104,7 +104,8 @@ def _init_dist_pytorch(backend, **kwargs) -> None:
             **kwargs)
     else:
         num_gpus = torch.cuda.device_count()
-        torch.cuda.set_device(rank % num_gpus)
+        local_rank = os.environ.get('LOCAL_RANK', None) or rank % num_gpus
+        torch.cuda.set_device(local_rank)
         torch_dist.init_process_group(backend=backend, **kwargs)
```

Comment:
torch.cuda.set_device() need a int parameter, but os.environ.get("LOCAL_RANK") returned a string, a type conversion is needed.

Reply:
Good point